imports/plugins/included/taxes-avalara/client/components/avalaraSettingsForm.js

```diff
@@ -0,0 +1,182 @@
+
+  // helper adds a class to every grid row
+  static customRowMetaData = {
+    bodyCssClassName: () =>  {
+      return "log-grid-row";
+    }
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      editingId: undefined,
+      showLogs: this.props.settings.avalara.enableLogging,
+      log: undefined,
+      fieldsProp: assign(clone(props.fieldsProp), {
+        "settings.avalara.enableLogging": {
+          handleChange: this.handleLogToggle
+        }
+      })
+    };
+  }
+
+  handleLogToggle = () => {
+    this.setState(() => ({ showLogs: !this.state.showLogs }));
+  }
+
+  handleTestCredentials = (event) => {
+    this.props.handleTestCredentials(event, this.props.settings);
+  }
+
+  editRow = (options) => {
+    const currentId = this.state.editingId;
+    this.setState(() => {
+      const id = options.props.data._id;
+      const log = Logs.findOne(id) || {};
+      log.data = JSON.stringify(log.data, null, 4);
+      return {
+        editingId: id,
+        log
+      };
+    });
+    // toggle edit mode clicking on same row
+    if (currentId === options.props.data._id) {
+      this.setState({
+        editingId: undefined,
+        log: undefined
+      });
+    }
+  }
+
+  render() {
+    const { handleSubmit, hiddenFields, settings, shownFields, shownLogFields } = this.props;
+
+    // add i18n handling to headers
+    const customColumnMetadata = AvalaraSettingsForm.filteredFields.reduce((arr, field) => {
+      arr.push({
+        accessor: field,
+        Header: i18next.t(`logGrid.columns.${field}`)
+      });
+      return arr;
+    }, []);
+
+
+    return (
+      <div className="rui avalara-update-form">
+        {!settings.avalara.apiLoginId &&
+          <div className="alert alert-info">
+            <Components.Translation defaultValue="Add API Login ID to enable" i18nKey="admin.taxSettings.avalaraCredentials" />
+            <a href="https://admin-development.avalara.net" target="_blank"> Avalara</a>
+          </div>
+        }
+        <Form
+          schema={AvalaraPackageConfig}
+          doc={{ settings }}
+          renderFromFields={true}
+          fieldsProp={this.state.fieldsProp}
+          docPath={"settings.avalara"}
+          name={"settings.avalara"}
+          fields={shownFields}
+          hideFields={hiddenFields}
+          onSubmit={handleSubmit}
+        />
+        <div id="testAvalaraCredentialsContainer">
+          <Components.Button id="testAvalaraCredentials" label="Test Credentials" buttonType="button"
+            className="btn btn-default" i18nKeyLabel="admin.dashboard.avalaraTestCredentials"
+            bezelStyle="outline" onClick={this.handleTestCredentials}
+          />
+        </div>
+        <div className="panel-body text-center avalara-login-box">
+          <a href="https://admin-development.avalara.net" target="_blank">Avalara Admin Console Login</a>
+        </div>
+        {this.state.showLogs &&
```

Comment:
In JSX, you need to make sure each condition is exactly `true` or `false` rather than just truthy or falsy. So always use `!!`. (`{!!this.state.showLogs &&`)